rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -0,0 +1,1173 @@
+        }
+
+        let timeseries_id =
+            generate_timeseries_id(from.org_id, from.project_id, from.metric_id, &from.tags);
+
+        let (tag_keys, tag_values): (Vec<_>, Vec<_>) = from.tags.into_iter().unzip();
+
+        let mut granularities = vec![
+            GRANULARITY_ONE_MINUTE,
+            GRANULARITY_ONE_HOUR,
+            GRANULARITY_ONE_DAY,
+        ];
+
+        if from.aggregation_option.unwrap_or_default() == "ten_second" {
+            granularities.push(GRANULARITY_TEN_SECONDS);
+        }
+
+        let count_value = match from.value {
+            MetricValue::Counter(value) => value,
+            _ => {
+                return Err(anyhow::anyhow!(
+                    "Unsupported values provided for counter metric type"
+                ))
+            }
+        };
+
+        Ok(Self {
+            use_case_id: from.use_case_id,
+            org_id: from.org_id,
+            project_id: from.project_id,
+            metric_type: "counter".to_string(),
+            metric_id: from.metric_id,
+            timestamp: from.timestamp,
+            retention_days: from.retention_days,
+            tags_key: tag_keys.iter().map(|k| k.parse::<u64>().unwrap()).collect(),
+            tags_indexed_value: vec![0; tag_keys.len()],
+            tags_raw_value: tag_values,
+            count_value,
+            materialization_version: 2,
+            timeseries_id,
+            granularities,
+            min_retention_days: None,
+            decasecond_retention_days: None,
+            hr_retention_days: None,
+            day_retention_days: None,
+            ..Default::default()
+        })
+    }
+}
+
+pub fn process_counter_message(
+    payload: KafkaPayload,
+    metadata: KafkaMessageMetadata,
+    config: &ProcessorConfig,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+    let msg: FromGenericMetricsMessage = serde_json::from_slice(payload_bytes)?;
+
+    let sentry_received_timestamp =
+        DateTime::from_timestamp(msg.sentry_received_timestamp.try_into().unwrap(), 0);
+
+    let result: Result<CountersRawRow, anyhow::Error> = msg.try_into();
+    match result {
+        Ok(mut row) => {
+            row.partition = metadata.partition;
+            row.offset = metadata.offset;
+            row.retention_days = enforce_retention(Some(row.retention_days), &config.env_config);
+
+            Ok(InsertBatch {
+                rows: RowData::from_rows([row])?,
+                origin_timestamp: None,
+                sentry_received_timestamp,
+            })
+        }
+        Err(err) => {
+            // If we get an error here, we should check if it is because of an unsupported
+            // metric type. If so, we should call InsertBatch::skip. Otherwise, we should
+            // return the original error.
+            if err.to_string().contains("Unsupported metric type") {
+                return Ok(InsertBatch::skip());
+            }
```

Comment:
how about returning an `Option<CountersRawRow>` instead and not treating this case as an error